<!DOCTYPE html>
<html class="no-js" lang="en">
	<head>
		<meta charset="utf-8">

		<!-- Always force latest IE rendering engine (even in intranet) & Chrome Frame
		Remove this if you use the .htaccess -->
		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

		<title>tRNAscan-SE Search Server</title>
		<meta name="description" content="Genomic tRNA database 2.0">
		<meta name="author" content="Patricia Chan">

		<meta name="viewport" content="width=device-width; initial-scale=1.0">

		<!-- Replace favicon.ico & apple-touch-icon.png in the root of your domain and delete these references -->
		<link rel="shortcut icon" href="/favicon.ico">
		<link rel="apple-touch-icon" href="/apple-touch-icon.png">
		<link rel="stylesheet" href="style/normalize.css">
		<link rel="stylesheet" href="style/foundation.css">
		<link rel="stylesheet" href="style/trnascanse.css">
		<link rel="stylesheet" id="font-awesome-css" href="style/font-awesome-4.3.0/css/font-awesome.min.css">
		<script src="js/vendor/modernizr.js"></script>
		<script src="js/vendor/jquery.js"></script>
  	</head>

	<body>

<!--small off canvas menu-->
<div class="off-canvas-wrap" data-offcanvas>
	<div class="inner-wrap">

<!--top menu-->
<nav class="tab-bar show-for-small">
	<a class="left-off-canvas-toggle menu-icon ">
		<span>Home</span>
	</a>  
</nav>

<nav class="top-bar hide-for-small" data-topbar>
	<ul class="title-area">
		<li class="name">
			<h1><a href="index.html">Home</a></h1>
		</li>
	</ul>

	<section class="top-bar-section">
		<ul class="right">
			<li class="divider"></li>
			<li class="has-dropdown">
				<a href="help.html" class="">Help</a>
			</li>
			<li class="divider"></li>
			<li class="has-dropdown">
				<a href="other-links.html" class="">Links</a>
				<ul class="dropdown">
					<li><a href="other-links.html">Other Links</a></li>
					<li><a href="http://gtrnadb.ucsc.edu/" target=_blank>GtRNAdb</a></li>
				</ul>
			</li>
			<li class="divider"></li>
			<li class="has-dropdown">
				<a href="about.html" class="">About</a>
				<ul class="dropdown">
					<li><a href="citation.html">Citation</a></li>
					<li><a href="contact.html">Contact Us</a></li>
				</ul>
			</li>
		</ul>
	</section>
</nav>

<aside class="left-off-canvas-menu">

<ul class="off-canvas-list">
	<li><label class="first">tRNAscan-SE</label></li>
	<li><a href="index.html">Home</a></li>
</ul>

<hr>

<ul class="off-canvas-list">
    <li><label>Help</label></li>
    <li><a href="help.html">Help</a></li>
</ul>

<hr>

<ul class="off-canvas-list">
    <li><label>Links</label></li>
    <li><a href="other-links.html">Other Links</a></li>
    <li><a href="http://gtrnadb.ucsc.edu/" target=_blank>GtRNAdb</a></li>
</ul>

<hr>

<ul class="off-canvas-list">
    <li><label>About</label></li>
	<li><a href="citation.html">Citation</a></li>
    <li><a href="contact.html">Contact Us</a></li>
    <li><a href="http://lowelab.ucsc.edu" target=_blank>Lowe Lab</a></li>
</ul>

</aside>


<!--content-->
<!--title-->
<header id="page-header">	  
	<div class="small-10 medium-7 large-4 columns">
		<img src="image/tRNAscanSE_with_name.png"/>
	</div>
	<div class="small-10 medium-7 large-4 columns">
		<h5>Help</h5>
	</div>
</header>

<div class="scroll-top-wrapper ">
	<span class="scroll-top-inner">
		<i class="fa fa-2x fa-arrow-up"></i>
	</span>
</div>

<section id="page-section">	
	<div class="row">
		<div class="small-24 medium-12 large-12 columns">
<BR>
<h4>User Interface</h4>
<P>The tRNAscan-SE interface
consists of three major components:<br>
* Search mode selection<br>
* Query sequence selection<br>
* Configuration of search-mode and options for displaying results
</P>
<P>The search mode determines which probabilistic model to use in searches
&#8211; each model is based on tRNA training data from selected species or
phylogenetic groups (for example, mammals, yeasts, archaea). If no explicit
model for the species of interest is available in the user interface,
specifying either a general model or a model from a related species
generally yields good results. Different search modes can offer varying
speed and sensitivity.&nbsp; For tRNAscan-SE, the standard search modes
are fast and very sensitive in most instances. However, for short
sequences that contain extremely atypical tRNAs, the &#8220;Infernal without HMM filter&#8221; mode
offers slightly improved sensitivity.
</P>
<P>
Query sequence selection is used to specify the sequences to be
searched for tRNAs. Raw or formatted sequence data can be pasted
directly into the query sequence box or can be uploaded from a local
file. Links to <a href="ExampleSequences.fa">sample query sequence data</a>
are also available for demonstration purposes.
</P>
<P>
The server also has a set of program-specific search and output-display
options such as the tRNA isotype code. In addition, the server has an
adjustable cutoff score enabling tradeoffs between scan sensitivity and
specificity. In most cases, the default parameter choices will be
satisfactory and should be selected &#8211; especially by new users. However,
more experienced users are able to exert some control over the
program&#8217;s results by manipulating these parameters.<br>
</P>
<h4>Output format</h4>
<P>
The output for tRNAscan-SE
includes overall statistics as well as summary data for
the entire search. The summary data includes counts of the total number
of tRNAs found, the number of tRNA pseudogenes found, number of tRNAs
with introns and which anticodons were detected. Finally, the output
shows the predicted secondary structure for each identified
sequence.&nbsp; Clicking on the &#8220;View&#8221; button of Predicted
tRNA structure will provide a two-dimensional representation of the familiar tRNA clover-leaf
structure. The output also displays the overall length of the
sequence, the location of the anticodon and the overall tRNAscan-SE
score.
</P>
<P>
Each predicted tRNA (not using mitochondrial tRNA search mode) is further evaluated with isotype-specific models.
These 20+ models consist of the universal 20 amino acids, plus special models to discriminate initiator/formyl-methionine
from elongator methionine, one to identify prokaryotic Ile tRNAs genomically encoded with a CAT anticodon,
and one to recognize selenocysteine tRNAs. Alongside the predicted isotype based on the anticodon, the highest scoring
isotype-specific model is also reported; any disagreement between the two functional prediction methods is reported
for closer inspection. The scores of all isotype-specific models for each identified tRNA are shown for comparison.  
</P>
<P>
The resulting tRNAs identified in the input sequence can be searched against
the <a href="http://gtrnadb.ucsc.edu" target=_blank>GtRNAdb</a> by clicking on the &#8220;View&#8221; button of Similar tRNAs in GtRNAdb.
Links to identical or highly similar tRNA genes found in the database will be listed. If a UCSC Genome Browser exists for any
identical or close tRNA matches, a direct link is provided to those matches, enabling the user to examine
its genomic context and any additional information available in genome browser tracks. 
</P>

<h4>Further information</h4>
The tRNAscan-SE algorithm is described in:<br>
<br>
<P>Chan, P.P., Lin, B., and Lowe, T.M. tRNAscan-SE 2.0. (In Preparation)</br>
Lowe, T.M. and Eddy, S.R. (1997) 
<a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?tool=pubmed&pubmedid=9023104"
>tRNAscan-SE: a program for improved
detection of transfer RNA genes in genomic sequence. </A><I>Nucleic Acids Research</I>,
<B>25</B>, 955-964.
</P>
		</div>
	</div>
</section>
	
<section id="footer">
	<div class="row">
		<div class="small-12 columns">
			<p><a href="http://lowelab.ucsc.edu" target=_blank>The Lowe Lab</a>, Biomolecular Engineering, University of California Santa Cruz</p>
		</div>
	</div>
</section>

		<a class="exit-off-canvas"></a>
	</div>      
</div>
		
		<script src="js/backToTop.js"></script>
	  	<script src="js/foundation.min.js"></script>
	  	<script type="text/javascript">
	    $(document).foundation();
	  	</script>
	</body>
</html>
