<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>snoGPS README</title>
</head>
<body>
<h2>Using the snoGPS Web Server<br>
</h2>
<h4><span style="font-weight: normal;">The snoGPS server is accessed
via the Lowe Lab Webserver Interface at&nbsp;
http://lowelab.ucsc.edu/snoGPS/. </span><br
 style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: bold;">User Interface</span><br
 style="font-weight: bold;">
<br style="font-weight: normal;">
<span style="font-weight: normal;">The snoGPS interface consists of
four major components:</span><br style="font-weight: normal;">
<span style="font-weight: normal;">* Search mode selection</span><br
 style="font-weight: normal;">
<span style="font-weight: normal;">* Query sequence selection</span><br
 style="font-weight: normal;">
<span style="font-weight: normal;">* Target sequence selection</span><br
 style="font-weight: normal;">
<span style="font-weight: normal;">* Configuration of search-mode and
options for displaying results</span><br style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: normal;">The search mode determines which
probabilistic model to use in searches &#8211; each model is based on snoRNA
training data from selected species or phylogenetic groups (i.e.,
mammals, yeasts, archaea). If no explicit model for the species of
interest is available in the user interface, specifying either a
general model or a model from a related species generally yields good
results. Different search modes can offer varying speed and
sensitivity. For snoGPS, search mode selection involves specifying
either single hairpin (&#8220;one-stem&#8221;) mode or double hairpin (&#8220;two-stem&#8221;)
mode. Because eukaryotic H/ACA snoRNAs have two hairpins, two-stem mode
searches have greater specificity and therefore are usually preferable.
However, in archaea, several known H/ACA RNAs have only a single
hairpin and would be missed with two-stem mode. Moreover, in certain
eukaryotic species (e.g. S. cerevisiae), for some of the known snoRNAs,
one hairpin has a highly irregular structure, which would evade
snoGPS.&nbsp; In these situations, scans using the one-stem search mode
can be effective if the two-stem scanner does not yield results.</span><br
 style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: normal;">Query sequence selection is used to
specify the sequences to be searched for snoRNAs. Raw or formatted
sequence data can be pasted directly into the query sequence box or can
be uploaded from a local file. Links to sample query sequence data at <a
 href="/snoGPS/U65.raw">http://lowelab.ucsc.edu/snoGPS/U65.raw</a>
are also available for demonstration purposes . snoGPS also expects
&#8220;target sequences&#8221; &#8211; i.e. sequences that may base-pair and be modified
by the query snoRNA sequence. Preloaded target sequences may be chosen,
including rRNA from human, yeast, and other model organisms.&nbsp;
Alternatively, the user can specify a custom target RNA sequence. As
with the query sequence, a custom target sequence can be pasted into a
box in either raw or formatted form, or can be uploaded from a file.
When using a custom target sequence, by default, every uridine in the
sequence is treated as a potential target.&nbsp; Alternatively, the
user can specify a subset of the target sequence nucleotides by
uploading a custom &#8220;pseudouridylation file&#8221; that indicates which
nucleotides to use as target sites. A sample pseudouridylation file is
included on the server via <a
 href="/snoGPS/puSiteFile.html">http://lowelab.ucsc.edu/snoGPS/puSiteFile.html</a>.&nbsp;
When pseudouridylation positions are known, restricting the search
space to these known target sites has the advantage of decreasing
search time and the number of false positive &#8220;hits&#8221;.</span><br
 style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: normal;">The server also has a set of
program-specific search and output-display options such as limits on
the minimum number of base pairings in the guide region. In addition,
the server has an adjustable cutoff score enabling tradeoffs between
scan sensitivity and specificity. In most cases, the default parameter
choices will be satisfactory and should be selected &#8211; especially by new
users. However, more experienced users are able to exert some control
over the program&#8217;s results by manipulating these parameters.</span><br
 style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: bold;">Output formats</span><br
 style="font-weight: bold;">
<br style="font-weight: normal;">
<span style="font-weight: normal;">snoGPS output includes a summary
listing of the candidate hits followed by graphical representations of
their predicted secondary structure.&nbsp; The summary listing for each
candidate sequence includes:</span><br style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: normal;">* Query sequence name and snoRNA
start and end positions within the query sequence</span><br
 style="font-weight: normal;">
<span style="font-weight: normal;">* snoGPS score</span><br
 style="font-weight: normal;">
<span style="font-weight: normal;">* Target sequence name and target
uridine position </span><br style="font-weight: normal;">
<span style="font-weight: normal;">* Total number of base pairings and
mismatches in the guide region</span><br style="font-weight: normal;">
<span style="font-weight: normal;">* Number of base pairs in the
hairpin surrounding the guide region</span><br
 style="font-weight: normal;">
<span style="font-weight: normal;">* Annotation as to whether the guide
region is in the 5' or 3'&nbsp;&nbsp; hairpin </span><br
 style="font-weight: normal;">
<span style="font-weight: normal;">* The length and strand of the
candidate subsequence</span><br style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: normal;">snoGPS scores for known snoRNA
sequences for various species are available on the website for
comparison. The predicted secondary structure of the sequence is
depicted in two ways. First the entire sequence is displayed along with
an &#8220;annotation&#8221; sequence string identifying the various sequence
motifs. Then the snoRNA guide region along with its basepairing to the
predicted target sequence is shown in graphical format. </span><br
 style="font-weight: normal;">
<br style="font-weight: normal;">
<span style="font-weight: bold;">Sample (abbreviated) snoGPS Output</span><br
 style="font-weight: bold;">
<span style="font-weight: bold;"></span><br style="font-weight: normal;">
<span style="font-weight: normal;">&gt;YourSeq.3 33.73 (70-135) Cmpl:
H_sapiens_LSU.U4417 U65 Pairs: 11/0/10/6/c 66 nt (W)</span><br
 style="font-weight: normal;">
<span style="font-weight: normal; font-family: monospace;">CCCCAGCTTAGGAAACAGGGTTGTTCTTCATGTGGATGACTCTGTGCCGAAAGCATGGGAACAGCT
</span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">X12&nbsp;
56XLLLLLLI12345678I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
I87654321IRRRRR X65&nbsp; 21X AAAAAA</span><br
 style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
--- </span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
/&nbsp;&nbsp; \ </span><br
 style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
\&nbsp;&nbsp; / </span><br
 style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
G-C</span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
G.U</span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
G-C</span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
A-U</span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
C-G</span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
A-U</span><br style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">5'-(N7)UAGGAA&nbsp;&nbsp;&nbsp;
GCCGA(N11)ACA</span><br
 style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
||||||&nbsp;&nbsp;&nbsp; ||||| </span><br
 style="font-weight: normal; font-family: monospace;">
<span style="font-weight: normal; font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
AUCCUU CY CGGCU </span><br style="font-family: monospace;">
</h4>
<h4>Interpreting snoGPS scores:</h4>
Since snoGPS is a hybrid deterministic-probabalisitic model, its "bit
scores" do not have the direct probabalistic interpretation of a purely
stochastic model. In particular, snoGPS scores depend on the specific
model being used. By running snoGPS on known H/ACA snoRNAs with the
selected search model and run parameters in the species of interest,
the user can determine what range of scores to expect for actual H/ACA
sequences. For example, runnning the default 2-stem yeast model against
40 rRNA pseudouridylation sites for which snoGPS is able to make a
prediction yields the following scores (numbers add up to more than the
number of yeast H/ACA snoRNAs since some snoRNAs have more than one
target site):<br>
<br>
<br>
<table cellpadding="2" cellspacing="2" border="1"
 style="text-align: left; width: 50%;">
  <tbody>
    <tr>
      <td style="vertical-align: top;">Score Range<br>
      </td>
      <td style="vertical-align: top;">Number of known<br>
yeast snoRNAs<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">0 - 25<br>
      </td>
      <td style="vertical-align: top; text-align: center;">4<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">&nbsp;25 - 35<br>
      </td>
      <td style="vertical-align: top; text-align: center;">7<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">35 - 45</td>
      <td style="vertical-align: top; text-align: center;">18<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">45 - 55<br>
      </td>
      <td style="vertical-align: top; text-align: center;">8<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">55 - 65<br>
      </td>
      <td style="vertical-align: top; text-align: center;">3<br>
      </td>
    </tr>
  </tbody>
</table>
<br>
Next, the scores for 78 H/ACA RNAs found by snoGPS in humans are shown.
In this case, since there is no experimental verification available for
any target assignments, each snoRNA score listed is the highest one for
any known pseudouridylation site. These scores used the mammalian
2-stem model.<br>
<br>
<table style="text-align: left; width: 50%;" border="1" cellspacing="2"
 cellpadding="2">
  <tbody>
    <tr>
      <td style="vertical-align: top;">Score Range<br>
      </td>
      <td style="vertical-align: top;">Number of <br>
yeast snoRNAs<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">0 - 35<br>
      </td>
      <td style="vertical-align: top; text-align: center;">8<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">35 - 45</td>
      <td style="vertical-align: top; text-align: center;">28<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">45 - 55<br>
      </td>
      <td style="vertical-align: top; text-align: center;">34<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">55 - 65<br>
      </td>
      <td style="vertical-align: top; text-align: center;">8<br>
      </td>
    </tr>
  </tbody>
</table>
<br>
In both of these examples, one sees that most known snoRNAs are found
with scores greater than 35 bits, using the two stem scanner.<br>
<br>
Scores with the one-stem scanners are typically lower.&nbsp; For
example, when testing S. cerevisiae H/ACA snoRNAs with the one stem
scanner, one finds the following range of scores:<br>
<br>
<table cellpadding="2" cellspacing="2" border="1"
 style="text-align: left; width: 50%;">
  <tbody>
    <tr>
      <td style="vertical-align: top;">Score Range<br>
&nbsp;(Yeast 1 stem scanner)<br>
      </td>
      <td style="vertical-align: top;">Number of <br>
yeast snoRNAs<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">0 - 20<br>
      </td>
      <td style="vertical-align: top; text-align: center;">4<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">20 - 25</td>
      <td style="vertical-align: top; text-align: center;">10<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">25 - 30<br>
      </td>
      <td style="vertical-align: top; text-align: center;">9<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">30 - 35<br>
      </td>
      <td style="vertical-align: top; text-align: center;">13<br>
      </td>
    </tr>
    <tr>
      <td style="vertical-align: top; text-align: center;">35 - 40<br>
      </td>
      <td style="vertical-align: top; text-align: center;">2<br>
      </td>
    </tr>
  </tbody>
</table>
<br>
As can be seen, the one-stem scores are significantly lower; a cutoff
of around 20 is needed to detect nearly all the known yeast H/ACA
snoRNAs with the one stem scanner. <br>
<h4>snoGPS structure annotations:</h4>
Each snoGPS record begins with a header line, e.g.:<br>
<font size="-1" style="font-weight: bold;"><span
 style="font-family: courier new,courier,monospace;">&gt;YourSeq.3
33.73 (70-135) Cmpl: H_sapiens_LSU.U4417 U65 Pairs: 11/0/10/6/c 66 nt
(W)</span></font><br>
<br>
Data in the header line are: <br>
* sequence name, <br>
* snoGPS score, <br>
* coordinates of predicted snoRNA within the sequence, <br>
* predicted target uridine, <br>
* any annotation that may be associated with target, <br>
* the number of basepairings and mismatches between the target and the
guide region, <br>
* the number of basepairings in the upper (or 'internal") and lower (or
'external') stems adjacent to the target region<br>
* an 'h' or a 'c' indicating whether snoGPS found the guide region in
the 5' ( "h") or the 3' ("aCa") hairpin<br>
* predicted snoRNA length in nucleotides<br>
* and whether the candidate was found on the positive ('W') or
reverse-complement ('C') strand.<br>
<br>
The snoGPS output includes a predicted secondary structure for each
identified H/ACA snoRNA, for example:<br>
<br>
<span style="font-family: monospace;">CCCCAGCTTAGGAAACAGGGTTGTTCTTCATGTGGATGACTCTGTGCCGAAAGCATGGGAACAGCT
</span><br style="font-family: monospace;">
<span style="font-family: monospace;">X12&nbsp;
56XLLLLLLI12345678I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
I87654321IRRRRR X65&nbsp; 21X AAAAAA<br>
</span><br>
In this structure annotation,&nbsp; 'HHHHHH' and 'CCCCCC' represent H
or ACA motifs respectively.&nbsp; ('AAAAAA' may indicate either an H or
an ACA motif, for example if only a scan for only a single hairpin is
performed as in the example above). 'LLLL'and 'RRRR' are the two parts
of the rRNA guide-sequence motif.&nbsp; Bases annotated with 'X'and 'I'
or a single digit between a pair of X's or a pair of I's refer to the
lower&nbsp; (or 'external')&nbsp; and upper (or 'internal') stems of
either the 5' or 3' hairpins.&nbsp; The use of digits to annotate the
bases within a stem is intended to facilitate visualizing which bases
are predicted to pair in the stem. In cases of an internal mismatch
within a stem or guide region, the corresponding number or letter is
deleted from the annotation. <br>
<br>
In addition to the above annotation of the entire predicted secondary
structure, a graphical representation of the guide region and its
matching target region is also included.&nbsp; Here is the guide
representation output by snoGPS for the same sequence:<br>
<span style="font-family: monospace;"><br
 style="font-family: monospace;">
</span><span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
--- </span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
/&nbsp;&nbsp; \ </span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
\&nbsp;&nbsp; / </span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
G-C</span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
G.U</span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
G-C</span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
A-U</span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
C-G</span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
A-U</span><br style="font-family: monospace;">
<span style="font-family: monospace;">5'-(N7)UAGGAA&nbsp;&nbsp;&nbsp;
GCCGA(N11)ACA</span><br style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
||||||&nbsp;&nbsp;&nbsp; ||||| </span><br
 style="font-family: monospace;">
<span style="font-family: monospace;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
AUCCUU CY CGGCU <br>
<br>
<br>
</span>
<h4>Further information<br>
</h4>
The snoGPS algorithm is described in:<br>
&nbsp;&nbsp;&nbsp; Schattner P, Decatur WA,&nbsp; Davis C,&nbsp; Ares,
M,&nbsp; Fournier, MJ and Lowe, TM, (2004) "Genome-wide Searching for
Pseudouridylation
Guide snoRNAs: Analysis of the Saccharomyces cerevisiae genome&#8221;,
Nucleic
Acids Res vol. 32(14): pp. 4281-4296<span
 style="font-family: monospace;"><br>
</span><br>
Additional information can be found in the documentation to the
stand-alone version of the program available at:<br>
http://lowelab.ucsc.edu/software/snoGPS-0.2.tar.gz<br>
<br>
<br>
</body>
</html>
