<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>The Lowe Lab - Research</title>
<link href="style/lowelab.css" rel="stylesheet" type="text/css" />
</head>

<body class="mainFrame">

<div id="container">
  <div id="header"> 
  	<table width="100%" border="0" cellspacing="0" cellpadding="0">
  	<tr>
    	<td width="75%"><a href="/"><img src="images/logo.jpg" alt="The Lowe Lab" width="53%"/></a></td>
    	<td><a href="http://www.soe.ucsc.edu/" target="_blank"><img src="images/SOE.jpg" alt="Baskin School of Engineering" width="100%"/></a></td>
  	</tr>
	</table>
  </div>
  <div id="menu">
    <table border="0" cellspacing="0" cellpadding="0" width="100%">
      <tr>
        <td ><a href="/">Home</a></td>
        <td ><a href="research.html">Research</a></td>
        <td ><a href="people.html">People</a></td>
        <td ><a href="publications.html">Publications</a></td>
        <td width="23%"><a href="databases.html">Tools and Databases</a></td>
        <td ><a href="teaching.html">Teaching</a></td>
        <td ><a href="contact.html">Contact Us</a></td>
      </tr>
    </table>
  </div>
  <div id="researchSidebar">
  <table width="100%" cellpadding="5" cellspacing="0">
  <tr>
    <td colspan="2">Click on the images to enlarge</td>
    </tr>
  <tr>
    <td colspan="2"><a href="images/Archaea-23S.jpg"><img src="images/Archaea-23S-small.jpg" width="100%" alt="Archaeal 23S phylogenetic tree" /></a>Genome diversity in Archaea</td>
    </tr>
  <tr>
  <tr>
    <td><a href="images/pae-rpr.jpg"><img src="images/pae-rpr-small.jpg" width=100% alt="P. aerophilum RNase P RNA" /></a><em>Pyrobaculum aerophilum</em> <br />RNase P RNA</td>
    <td><a href="images/Ape-tRNA-Asp-2D.jpg"><img src="images/Ape-tRNA-Asp-2D-small.jpg" width="70%" alt="A. pernix pre-tRNA-Asp-GUC" /></a><br /><em>Trans</em>-spliced split tRNA in <em>Aeropyrum pernix</em></td>
  </tr>
  <tr>
    <td colspan="2"><img src="images/bottles.jpg" width="100%" alt="Culture bottles" /><em>Pyrobaculum</em> cultures grown at 95°C</td>
    </tr>
  <tr>
    <td colspan="2"><a href="images/CDBox.jpg"><img src="images/CDBox-small.jpg" width="100%"alt="C/D box snoRNA" /></a>Probabilistic non-coding RNA gene modeling</td>
    </tr>
  <tr>
    <td colspan="2"><img src="images/cloning.jpg" width="100%" alt="cell colonies" />Characterizing RNA transcripts</td>
    </tr>
  <tr>
    <td colspan="2"><a href="images/OctopusSpring.jpg"><img src="images/OctopusSpring-small.jpg" width="100%" alt="Octopus Spring" /></a>Octopus Spring in Yellowstone National Park</td>
    </tr>
</table>
  <!-- end #researchSidebar --></div>
  <div id="researchContent">
  	<div id="menuLink">
    <p><a href="/">Home</a> >> <a href="research.html">Research</a></p>
    </div>
    <h3>Non-coding RNAs</h3>
    <p>
   	RNase P is best known for its role in removing the 5′ leaders of pre-tRNAs, an essential step in tRNA maturation. The RNA component of this holoenzyme is thought to be a universal feature of life. The inability to identify RNase P in some organisms including <em>Pyrobaculum</em> has sown doubts about this phenomenon. Using comparative genomics and improved computational methods, we, in collaboration with Venkat Gopalan's lab and Jim Brown's lab, have now identified a radically <a href="http://www.pnas.org/content/107/52/22493.long" target="_blank">minimized form of the RNase P RNA</a> in five <em>Pyrobaculum</em> species and the related crenarchaea <em>Caldivirga maquilingensis</em> and <em>Vulcanisaeta distributa</em>, all retaining a conventional catalytic domain, but lacking a recognizable specificity domain. These “Type T” RNase P RNAs are the smallest naturally occurring form yet discovered to function as <em>trans</em>-acting ribozymes. Due to the absence of almost half of a typical gene, the archaeal RNase P RNA covariance search model fails to detect their existence. We then developed a type T specific covariance model that located shortened RNase P RNA in <em>Thermoproteus tenax</em>, <em>Thermoproteus uzoniensis</em>, and <em>Vulcanisaeta moutnovskia</em>.</p>
    <p>As in eukaryotes, pre-tRNAs in archaea often contain introns that are removed in tRNA maturation. Two unrelated archaeal species display unique pre-tRNA processing complexity in the form of “split” tRNA genes: 2-3 segments of tRNAs are transcribed from different loci, then <em>trans</em>-spliced to form a mature tRNA. Another rare type of pre-tRNA, found only in eukaryotic algae, is “permuted” where the 3′ half is encoded upstream of the 5′ half, and must be processed to be functional. 
    Using an improved version of the gene-finding program <a href="http://lowelab.ucsc.edu/tRNAscan-SE/" target="_blank">tRNAscan-SE</a>, comparative analyses, and experimental verifications, we have identified four novel <em>trans</em>-spliced tRNA genes, each in a different species of the Desulfurococcales branch of the Archaea. Additionally, we discovered the first examples of permuted tRNA genes in Archaea, which appear to be permuted in the same arrangement seen previously in red alga. These findings illustrate that split tRNAs are sporadically spread across a major branch of the Archaea, and that permuted tRNAs are a new shared characteristic between archaeal and eukaryotic species. The <a href="http://genomebiology.com/content/12/4/R38" target="_blank">split tRNA discoveries</a> also provide new clues to their evolutionary history, supporting hypotheses for recent acquisition via viral or other mobile elements. </p>
      <p>
      The box H/ACA RNA gene family is one of the largest non-protein-coding gene families in eukaryotes and archaea. We developed <a href="http://lowelab.ucsc.edu/snoGPS/" target="_blank">snoGPS</a>, a computational screening program for H/ACA snoRNAs, and initially applied it to <em>Saccharomyces cerevisiae</em>. We then extended our method to screen for H/ACA RNAs in  human, mouse, and rat genomes.  Complementing extensive cloning experiments performed by others, our findings include the detection and experimental verification of seven new mammalian H/ACA RNAs and the prediction of 23 new H/ACA RNA pseudouridine guide assignments. These assignments include four for H/ACA RNAs previously classified as orphan H/ACA RNAs with no known targets. We also determined systematic syntenic conservation among human and mouse H/ACA RNAs. Furthermore, 82 of 97 ribosomal RNA pseudouridines and 18 of 32 spliceosomal RNA pseudouridines in mammals have been linked to H/ACA guide RNAs. </p>
    <h3>Archaeal Biology</h3>
    <p>
    Many archaea live in hostile environments, from salt lakes to acidic hot springs, but they can be very difficult to grow and study in the laboratory. We teamed up with Chad Saltikov (METX of UCSC), Sorel Fitz-Gibbon (UCLA) and Christopher House (Pennsylvania State University), with the support of Joint Genome Institute, to sequence five closely-related heat-loving archaea (<em>Pyrobaculum arsenaticum</em>, <em>Pyrobaculum calidifontis</em>, <em>Pyrobaculum islandicum</em>, <em>Thermoproteus neutrophilus</em>, and <em>Caldivirga maquilingensis</em>) found in various hot springs from Iceland to the Philippines. These species belong to a unique group of archaea that can use sulfur compounds and toxic metals such as arsenic and selenium to power their cells. They use these compounds in much the same way that human cells use oxygen. These genomes are currently available at the <a href="http://archaea.ucsc.edu" target="_blank">UCSC Archaeal Genome Browser</a>.</p>
    <p>The mixture of eukaryotic and bacterial features in transcription and translation mechanisms also prompts the interest of better understanding the transcription unit structure in archaea. Existing computational algorithms for operon predictions are tailored and validated largely with bacterial training data, which may not be well applicable to archaea. Previous studies on archaeal transcriptional and translational signals limited the analyzed region and focused only on a small portion of genes in a genome. We extended the search area and identified promoters and Shine-Dalgarno motifs in 46 archaeal genomes. Besides recognizing the general patterns of transcript structure in these genomes, the analyzed results revealed that the domination of leadered and leaderless transcripts are equally spread in Crenarchaeota and Euryarchaeota, the two main archaeal phyla. We also identified and experimentally verified internal promoters within upstream coding regions, and uncovered Shine-Dalgarno-less leadered transcripts in <em>Pyrococcus furiosus</em>. These findings suggest the complexity of archaeal transcription units and serve as a base for novel gene finding including the elusive <em>Pyrobaculum</em> RNase P, and permuted and recently split tRNAs in archaea.</p>
<p>
  Hyperthermophilic crenarchaea in the genus <em>Pyrobaculum</em> are notable for respiratory versatility, but relatively little is known about the genetics or regulation of crenarchaeal respiratory pathways. We measured global <a href="http://jb.asm.org/cgi/content/full/191/3/782?view=long&amp;pmid=19047344" target="_blank">gene expression in <em>Pyrobaculum aerophilum</em> cultured with oxygen, nitrate, arsenate and ferric iron</a> as terminal electron acceptors to identify transcriptional patterns that differentiate these pathways. We also compared genome sequences for four closely related species with diverse respiratory characteristics (<em>Pyrobaculum arsenaticum</em>, <em>Pyrobaculum calidifontis</em>, <em>Pyrobaculum islandicum</em>, and <em>Thermoproteus neutrophilus</em>) to identify genes associated with different respiratory capabilities. Specific patterns of gene expression in <em>P. aerophilum</em> were associated with aerobic respiration, nitrate respiration, arsenate respiration, and anoxia. Functional predictions based on these patterns include separate cytochrome oxidases for aerobic growth and oxygen scavenging, a nitric oxide-responsive transcriptional regulator, a multicopper oxidase involved in denitrification, and an archaeal arsenate respiratory reductase. We were unable to identify specific genes for iron respiration, but <em>P. aerophilum</em> exhibited repressive transcriptional responses to iron remarkably similar to those controlled by the ferric uptake regulator in bacteria. Together, these analyses present a genome-scale view of crenarchaeal respiratory flexibility and support a large number of functional and regulatory predictions for further investigation.
</p>
	<!-- end #researchContent --><br class="clearfloat" /></div>
<!-- end #container --></div>
</body>
</html>
